daemon/container_operations_unix.go

```diff
@@ -120,19 +122,21 @@ func (daemon *Daemon) setupIpcDirs(c *container.Container) error {
 				return err
 			}
 
-			shmSize := int64(daemon.configStore.ShmSize)
-			if c.HostConfig.ShmSize != 0 {
-				shmSize = c.HostConfig.ShmSize
-			}
-			shmproperty := "mode=1777,size=" + strconv.FormatInt(shmSize, 10)
+			shmproperty := "mode=1777,size=" + strconv.FormatInt(c.HostConfig.ShmSize, 10)
 			if err := unix.Mount("shm", shmPath, "tmpfs", uintptr(unix.MS_NOEXEC|unix.MS_NOSUID|unix.MS_NODEV), label.FormatMountLabel(shmproperty, c.GetMountLabel())); err != nil {
 				return fmt.Errorf("mounting shm tmpfs: %s", err)
 			}
 			if err := os.Chown(shmPath, rootIDs.UID, rootIDs.GID); err != nil {
 				return err
 			}
+			c.ShmPath = shmPath
 		}
 
+	case ipcMode.IsPrivate():
+		c.ShmPath = ""
+
+	default:
```

Comment:
We do have `docker update`, but it's a curated list of options that we allow updating; generally, containers should be considered immutable, so re-creating the container is in most cases the answer. For `docker update` we (so far) limited options to updating resource-constraints, e.g. update memory-limits for an existing container (as that can be considered an operational concern)